hi-ml-multimodal/test_multimodal/image/model/test_model.py

```diff
@@ -159,6 +159,81 @@ def test_hubconf() -> None:
         assert torch.allclose(value_hub, value_himl)
 
 
+def test_load_biovil_weights() -> None:
+    """Test that the ``biovil`` option loads weights correctly."""
+    expected_model = get_biovil_resnet()
+
+    joint_feature_size = 128
+    biovil_model = ImageModel(img_model_type="resnet50",
+                              joint_feature_size=joint_feature_size,
+                              weights="biovil")
+
+    for biovil_param, expected_param in zip(
+        biovil_model.named_parameters(),
+        expected_model.named_parameters()
+    ):
+        assert biovil_param[0] == expected_param[0]
+        assert torch.isclose(biovil_param[1], expected_param[1]).all()
+
+
+def test_load_imagenet_weights() -> None:
+    """Test that the ``imagenet`` option loads weights correctly."""
+    expected_model = resnet50(pretrained=True)
+
+    joint_feature_size = 4
+    imagenet_model = ImageModel(img_model_type="resnet50",
+                                joint_feature_size=joint_feature_size,
+                                weights="imagenet")
+
+    for imagenet_param, expected_param in zip(
+        imagenet_model.encoder.encoder.named_parameters(),
+        expected_model.named_parameters()
+    ):
+        assert imagenet_param[0] == expected_param[0]
+        assert torch.isclose(imagenet_param[1], expected_param[1]).all()
+
+
+def test_load_random_weights() -> None:
+    """Test that the ``random`` option loads weights correctly."""
+    torch.manual_seed(5)
+    imagenet_model = resnet50(pretrained=True)
+    biovil_model = get_biovil_resnet()
+    joint_feature_size = 128
+
+    # Compare random weights with ImageNet and BioViL weights
+    random_model_1 = ImageModel(img_model_type="resnet50",
+                                joint_feature_size=joint_feature_size,
+                                weights="random")
+
+    for random_param_1, imagenet_param in zip(
+        random_model_1.encoder.encoder.named_parameters(),
+        imagenet_model.named_parameters()
+    ):
+        assert random_param_1[0] == imagenet_param[0]
+        assert not torch.isclose(random_param_1[1], imagenet_param[1]).all()
+
+    for random_param_1, biovil_param in zip(
+        random_model_1.named_parameters(), biovil_model.named_parameters()
+    ):
+        assert random_param_1[0] == biovil_param[0]
+        assert not torch.isclose(random_param_1[1], biovil_param[1]).all()
+
+    # Compare two random model weights
+    random_model_2 = ImageModel(img_model_type="resnet50",
+                                joint_feature_size=joint_feature_size,
+                                weights="random")
+
+    for random_param_1, random_param_2 in zip(
+        random_model_1.encoder.encoder.named_parameters(),
+        random_model_2.encoder.encoder.named_parameters()
+    ):
+        assert random_param_1[0] == random_param_2[0]
+
+        # BN and down_sample initialisations are the same in random models
```

Comment:
Do you know why they are the same?

Reply:
I assumed batchnorm was the same as in Tensorflow, where beta and gamma are initialised to 0 and 1 respectively, but I'll check

Reply:
Ah that would make sense! Thanks, I was curious.